src/providers/stage-providers.hpp

```diff
@@ -53,19 +53,14 @@ std::unique_ptr<EdgeDetectionAlgorithm> ProvideEdgeDetectionAlgorithm(DistanceOp
  * 
  * @return std::unique_ptr<DistanceDeterminationAlgorithm> The distance determination algorithm
  */
-std::unique_ptr<DistanceDeterminationAlgorithm> ProvideDistanceDeterminationAlgorithm(DistanceOptions &&options) {
+inline std::unique_ptr<DistanceDeterminationAlgorithm> ProvideDistanceDeterminationAlgorithm(const DistanceOptions &options) {
     if (options.distanceAlgo == SDDA) {
-        return std::make_unique<SphericalDistanceDeterminationAlgorithm>(options.radius,
```

Comment:
There's actually a point to constructing it this way. Construction directly in the constructor allows the function to save memory, instead of constructing it in the function and then copying it.

C++ is funny like that, it doesn't optimize things well sometimes unless you tell it too.